src/openrct2/ride/Ride.cpp

```diff
@@ -4505,8 +4503,8 @@ static Vehicle* vehicle_create_car(
         }
         vehicle->TrackSubposition = subposition;
 
-        vehicle->track_x = x;
-        vehicle->track_y = y;
+        vehicle->TrackLocation.x = x;
```

Comment:
```suggestion
        vehicle->TrackLocation = {x, y, tileElement->GetBaseZ()};
```
`tileElement` does not change afterwards, so you can move it up here.